src/map/utils/diggingutils.cpp

```diff
@@ -0,0 +1,490 @@
+			m_MessageOffset[2][zone] = luautils::GetTextIDVariable(zone, "ITEM_OBTAINED");
+		}
+	}
+
+/************************************************************************
+*																		*
+*	Chocobo Digging Action												*
+*																		*
+************************************************************************/
+
+	void Dig(CCharEntity* PChar)
+	{
+		int8 PlayerCount = GetPlayerDigCount(PChar);
+		uint16 MessageOffset = m_MessageOffset[0][PChar->getZone()];
+		uint16 MessageOffset2 = m_MessageOffset[1][PChar->getZone()];
+		uint16 MessageOffset3 = m_MessageOffset[2][PChar->getZone()];
+
+		int8 Chance = WELL512::irand() % 100;
+
+		if (Chance <= 15)	// Check if we lose chance
+		{
+			PChar->pushPacket(new CMessageTextPacket(PChar, MessageOffset2));
+			CalculateSkillUp(PChar);
+		}
+		else				// Calculate item received and display message
+		{
+			int32 ItemID = 4545; // Temporary item for testing purposes, will give Gysahl Greens (to be removed before committing)
+			DigList_t* DigList = itemutils::GetDigList(PChar->getZone());
+			if (DigList != NULL && DigList->size())
+			{
+				uint16 random = WELL512::irand()%1000;
+				uint8 randItem = WELL512::irand()%DigList->size();
+				if(random < (DigList->at(randItem).DigRate + map_config.dig_rate_bonus))
+				{
+					// Grab a randomly item from the list for the zone
+					// Check the requirements and then give the item if they are met
+					if(DigList->at(randItem).DigRequirements == 0)
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 1 && CanPlayerBurrow(PChar) == true) // Burrow
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 2 && CanPlayerBore(PChar) == true) // Bore
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 3 && EggHelm(PChar) == true) // Egg Helm
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 4 && NightDigging() == true) // Nighttime
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					ShowDebug(CL_CYAN"Item ID Dug up %d \n", ItemID);
+				}
+				if (ItemID != 0)
+				{
+					if (PChar->getStorage(LOC_INVENTORY)->GetFreeSlotsCount() != 0)
+					{
+						// Give the item to the player
+						charutils::AddItem(PChar, LOC_INVENTORY, ItemID);
+						PChar->pushPacket(new CMessageSpecialPacket(PChar, MessageOffset3, ItemID, 0, 0));
+						UpdateZoneItemsDug(PChar);
+					}
+					else
+					{
+						// Inventory is full, throw item away
+						PChar->pushPacket(new CMessageSpecialPacket(PChar, MessageOffset, ItemID, ItemID, 0));
+					}
+				}
+				else
+				{
+					// Beat the Dig chance but not the Item chance
+					PChar->pushPacket(new CMessageTextPacket(PChar, MessageOffset2));
+				}
+			}
+			CalculateSkillUp(PChar);
+		}
+		UpdatePlayerDigCount(PChar);
+		SetLastDig(PChar);
```

Comment:
stick these two in the same function (UpdatePlayerDigCount), not really a need to separate them since they're getting updated at the same time